File: examples/python/evm/account.list_token_balances.py
# Usage: uv run python evm/account.list_token_balances.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.get_or_create_account(name="Account1")
        print(f"Account: {account.address}")

        print("Listing token balances for account...")
        balances = await account.list_token_balances(
            network="base-sepolia",
            page_size=10,
        )
        print(f"Balances: {balances}")


asyncio.run(main())

File: examples/typescript/evm/account.listTokenBalances.ts
// Usage: pnpm tsx evm/account.listTokenBalances.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import { createPublicClient, http } from "viem";
import { baseSepolia } from "viem/chains";
import "dotenv/config";

const cdp = new CdpClient();

const account = await cdp.evm.getOrCreateAccount({ name: "MyAccount" });

const { transactionHash } = await account.requestFaucet({
  network: "base-sepolia",
  token: "eth",
});

await createPublicClient({
  transport: http(),
  chain: baseSepolia,
}).waitForTransactionReceipt({
  hash: transactionHash,
});

const result = await account.listTokenBalances({
  network: "base-sepolia",
});

for (const balance of result.balances) {
  console.log("Token contract address:", balance.token.contractAddress);
  console.log("Token balance:", balance.amount.amount);
}
